src/virtualenv/create/via_global_ref/builtin/cpython/mac_os.py

```diff
@@ -176,6 +175,33 @@ def fix_mach_o(exe, current, new, max_size):
             raise
 
 
+def fix_signature(exe):
```

Comment:
I've restructured it to introduce a **CPython2macOsArmFramework** class, and have it just passing in the python interpreter to sign.
It was tested on the M1 and an existing Darwin intel machine.
In the latest commit I've also removed **can_create** from  CPython2macOsFramework since it is now only calling the parent function.

Reply:
Can you inline this into the create above, otherwise looks :+1:

Reply:
I've inlined the function into the arm specific class